diff --git a/ios/SecureStoreModule.swift b/ios/SecureStoreModule.swift
index 7af7fe54431edbdf4be6ffecaa5b17f90deebbf8..2a55fc26393e85ebc773be11d77bcc802de59579 100644
--- a/ios/SecureStoreModule.swift
+++ b/ios/SecureStoreModule.swift
@@ -96,7 +96,7 @@ public final class SecureStoreModule: Module {
       guard let _ = Bundle.main.infoDictionary?["NSFaceIDUsageDescription"] as? String else {
         throw MissingPlistKeyException()
       }
-      let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, SecAccessControlCreateFlags.biometryCurrentSet, nil)
+      let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, SecAccessControlCreateFlags.userPresence, nil)
       setItemQuery[kSecAttrAccessControl as String] = accessOptions
     }
 
