diff --git a/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt b/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt
index a281b88ec7d9d19870c8707347251cac2137b4c5..07e73a591309a431364168b61422efd32978d037 100644
--- a/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt
+++ b/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt
@@ -58,8 +58,10 @@ class AuthenticationHelper(
 
   fun assertBiometricsSupport() {
     val biometricManager = BiometricManager.from(context)
+    val biometricAuthenticators = BiometricManager.Authenticators.BIOMETRIC_STRONG or
+                                  BiometricManager.Authenticators.DEVICE_CREDENTIAL;
     @SuppressLint("SwitchIntDef") // BiometricManager.BIOMETRIC_SUCCESS shouldn't do anything
-    when (biometricManager.canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_STRONG)) {
+    when (biometricManager.canAuthenticate(biometricAuthenticators)) {
       BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE, BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE -> {
         throw AuthenticationException("No hardware available for biometric authentication. Use expo-local-authentication to check if the device supports it")
       }
diff --git a/android/src/main/java/expo/modules/securestore/AuthenticationPrompt.kt b/android/src/main/java/expo/modules/securestore/AuthenticationPrompt.kt
index 7437374f164c7c7a47b7cb78fffb374cf52b6905..92a07c290f296c8fcabce64e761ae39fce0b8895 100644
--- a/android/src/main/java/expo/modules/securestore/AuthenticationPrompt.kt
+++ b/android/src/main/java/expo/modules/securestore/AuthenticationPrompt.kt
@@ -1,6 +1,7 @@
 package expo.modules.securestore
 
 import android.content.Context
+import androidx.biometric.BiometricManager
 import androidx.biometric.BiometricPrompt
 import androidx.biometric.BiometricPrompt.PromptInfo
 import androidx.core.content.ContextCompat
@@ -15,7 +16,10 @@ class AuthenticationPrompt(private val currentActivity: FragmentActivity, contex
   private var executor: Executor = ContextCompat.getMainExecutor(context)
   private var promptInfo = PromptInfo.Builder()
     .setTitle(title)
-    .setNegativeButtonText(context.getString(android.R.string.cancel))
+    .setAllowedAuthenticators(
+      BiometricManager.Authenticators.BIOMETRIC_STRONG or
+      BiometricManager.Authenticators.DEVICE_CREDENTIAL
+    )
     .build()
 
   suspend fun authenticate(cipher: Cipher): BiometricPrompt.AuthenticationResult? =
diff --git a/android/src/main/java/expo/modules/securestore/encryptors/AESEncryptor.kt b/android/src/main/java/expo/modules/securestore/encryptors/AESEncryptor.kt
index 3a12dc996a27c1d97ac110d808f1cdf98e5379b9..d76addfd8d616a2e8d8dae34899369719da89ea7 100644
--- a/android/src/main/java/expo/modules/securestore/encryptors/AESEncryptor.kt
+++ b/android/src/main/java/expo/modules/securestore/encryptors/AESEncryptor.kt
@@ -61,6 +61,7 @@ class AESEncryptor : KeyBasedEncryptor<KeyStore.SecretKeyEntry> {
       .setBlockModes(KeyProperties.BLOCK_MODE_GCM)
       .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
       .setUserAuthenticationRequired(options.requireAuthentication)
+      .setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG or KeyProperties.AUTH_DEVICE_CREDENTIAL)
       .build()
 
     val keyGenerator = KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES, keyStore.provider)
diff --git a/ios/SecureStoreModule.swift b/ios/SecureStoreModule.swift
index 584df218365e2ebf176f6d332970a3ba64d7a12e..83b7eeb4fa62488cf0c7ccfe8889553ed1ee1aaa 100644
--- a/ios/SecureStoreModule.swift
+++ b/ios/SecureStoreModule.swift
@@ -102,7 +102,7 @@ public final class SecureStoreModule: Module {
       }
 
       var error: Unmanaged<CFError>? = nil
-      guard let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, .biometryCurrentSet, &error) else {
+      guard let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, .userPresence, &error) else {
         let errorCode = error.map { CFErrorGetCode($0.takeRetainedValue()) }
         throw SecAccessControlError(errorCode)
       }
